circle/functions

```diff
@@ -581,28 +604,120 @@ update_chart_in_repo() {
 
       # auto merge updates to https://github.com/bitnami/charts
       if [[ $REPO_TO_UPDATE == "https://github.com/bitnami/charts" ]]; then
-        info "Auto-merging $CHART_NAME-$CHART_VERSION_FIRST_UPDATE..."
-        git checkout master >/dev/null
-        git merge --no-ff $CHART_NAME-$CHART_VERSION_FIRST_UPDATE >/dev/null
-        git remote remove origin >/dev/null
-        git remote add origin https://$GITHUB_USER@$(echo ${REPO_TO_UPDATE/https:\/\/}).git >/dev/null
-        git push origin master >/dev/null
+        git_automerge_branch $REPO_TO_UPDATE "${CHART_NAME}-${CHART_VERSION_FIRST_UPDATE}"
       fi
     fi
 
     info "Cleaning up old branches..."
-    git fetch development >/dev/null
-    for branch in $(git branch --remote --list development/$CHART_NAME-* | sed 's?.*development/??' | grep -v "^$CHART_NAME-$CHART_VERSION_FIRST_UPDATE$")
-    do
-      log "Deleting $branch..."
-      git push development :$branch >/dev/null
-    done
+    git_cleanup_old_branches "${CHART_NAME}-*" "^$CHART_NAME-$CHART_VERSION_FIRST_UPDATE$"
   else
     warn "Chart release/updates skipped!"
   fi
-  
+
   info "Cleaning up repository"
   # Exiting from charts/ repository
   cd ..
   rm -rf charts
 }
+
+# Expects three parameters: <repo to update> <distro codename> and <docker image digest>
+update_minideb_derived() {
+    if [ -z "$1" ] || [ -z "$2" ] || [ -z "$3" ]
+    then
+        error "Wrong arguments passed to update_minideb_extras function"
+        exit 1
+    fi
+
+    REPO_TO_UPDATE=$1
+    DIST=$2
+    DIST_BUILD_ID=$3
+    BASENAME=bitnami/minideb
+
+    echo "Cloning '$REPO_TO_UPDATE' repo..."
+    if ! git clone --quiet --single-branch $REPO_TO_UPDATE minideb-extras; then
+        error "Could not clone $REPO_TO_UPDATE..."
+        exit 1
+    fi
+    cd minideb-extras
+
+    # add development remote
+    git remote add development "https://$GITHUB_USER@github.com/$GITHUB_USER/$(echo ${REPO_TO_UPDATE/https:\/\/github.com\/bitnami\/} | tr / -).git"
+
+    # lookup the distro in minideb-extras
+    DIST_PATH=
+    for d in $(find * -type d -name $DIST )
+    do
+        if [ -f "$d/Dockerfile" ]; then
+            DIST_PATH=$d
+            break
+        fi
+    done
+
+    if [[ -z $DIST_PATH ]]; then
+        error "Distro '$DIST' could not be found in '$REPO_TO_UPDATE' repo"
+        continue
+    else
+        cd "$DIST_PATH"
+    fi
+
+    if [[ -z $GITHUB_USER || -z $GITHUB_PASSWORD ]]; then
```

Comment:
I guess you can move this check to the beginning of the function. Note that you are already trying to use `$GITHUB_USER` and `$GITHUB_PASSWORD` when adding the development branch

Reply:
Fixed